annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastProcessor.java

```diff
@@ -82,15 +87,60 @@ private void reportDuplicates(String fastTypeName, Collection<Element> elements)
     }
 
     private void buildCreatorClass(HashMap<String, Element> nameClassMap) throws IOException {
-        TypeName typeName = new TypeName("com.exactpro.epfast.annotation.internal.CreatorImpl");
+        TypeName provider = new TypeName("com.exactpro.epfast.annotation.internal.CreatorImpl");
+        TypeName service = new TypeName("com.exactpro.epfast.ICreator");
         // mustache files are in UTF-8 by default
         Mustache mustache = mustacheFactory.compile(
             "com/exactpro/epfast/annotation/processing/CreatorImpl.java.mustache");
-
-        JavaFileObject builderFile = processingEnv.getFiler().createSourceFile(typeName.toString());
+        Collection<Element> elements = nameClassMap.values();
+        addServiceEntry(service.toString(), provider.toString(), elements.toArray(new Element[0]));
+        JavaFileObject builderFile = processingEnv.getFiler().createSourceFile(provider.toString());
         try (PrintWriter out = new PrintWriter(builderFile.openWriter())) {
-            mustache.execute(out, new CreatorTemplateParameters(typeName, nameClassMap.entrySet())).flush();
+            mustache.execute(out, new CreatorTemplateParameters(provider, nameClassMap.entrySet())).flush();
+        }
+    }
+
+    private void addServiceEntry(String service, String provider, Element... originatingElements) throws IOException {
```

Comment:
I forgot we have `TypeName`. It's better to use it instead of String for service and provider arguments